### PR TITLE
Feature/remove gradle license report plugin

### PR DESCRIPTION
**Pull request by andsel**
## Release notes

[rn:skip]

## What does this PR do?

Switch from `Gradle-License-Report` to a custom task to list project dependencies.
The Gradle-License-Report was use to generate a license report in CSV with many columns, but the only one used by the [dependency report tool](https://github.com/elastic/logstash/blob/85abb95d9a3818fdbb40be2ed92cfd4c2fe11c8a/tools/dependencies-report/src/main/java/org/logstash/dependencies/Dependency.java#L96) is the first, containing the full artifact name.

The plugin is not yet compatible with Gradle 7 and it's an external dependency to keep update. The to job to list all jar dependencies could be done with a simple Gradle task, this PR introduces this custom task to be used in license verification flow.

## Why is it important/What is the impact to the user?

It doesn't impact directly the user.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] ~~I have made corresponding changes to the documentation~~
- [x] I have made corresponding change to the default configuration files (and/or docker env variables)
- [x] I have added tests that prove my fix is effective or that my feature works

## Author's Checklist

- [x] Add a dependency on `logstash-core` not listed in notices and verify it's identified as a violation by `ci/license_check.sh`

## How to test this PR locally

Add a new dependency to a subproject, for example `logstash-core`, verify it's identified as a violation by `ci/license_check.sh`

## Related issues

- Relates #13177

### CODE REVIEW

File: buildSrc/src/main/groovy/org/logstash/gradle/tooling/ListProjectDependencies.groovy [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+    @OutputFile
+    File reportFile = project.file("${project.buildDir}/reports/dependency-license/" + report)
+
+    @OutputDirectory
+    File reportDir = reportFile.toPath().parent.toFile()
+
+    ListProjectDependencies() {
+        description = "Lists the projects dependencies in a CSV file"
+        group = "org.logstash.tooling"
```
Comment (robbavey) [inline buildSrc/src/main/groovy/org/logstash/gradle/tooling/ListProjectDependencies.groovy:25]: Is it relevant that this group is different to the package name?

File: tools/dependencies-report/src/main/resources/licenseMapping.csv [hunk 2]
```diff
@@ -23,6 +23,7 @@ dependency,dependencyUrl,licenseOverride,copyright,sourceURL
 "com.fasterxml.jackson.core:jackson-databind:",https://github.com/FasterXML/jackson-databind,Apache-2.0
 "com.fasterxml.jackson.core:jackson-core:",https://github.com/FasterXML/jackson-core,Apache-2.0
 "com.fasterxml.jackson.dataformat:jackson-dataformat-cbor:",https://github.com/FasterXML/jackson-dataformats-binary,Apache-2.0
+"com.fasterxml.jackson.dataformat:jackson-dataformat-yaml:",https://github.com/FasterXML/jackson-dataformats-binary,Apache-2.0
```
Comment (robbavey) [inline tools/dependencies-report/src/main/resources/licenseMapping.csv:26]: Do we know why these new dependencies showed up?

File: buildSrc/src/main/groovy/org/logstash/gradle/tooling/ListProjectDependencies.groovy [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+import org.gradle.api.tasks.OutputDirectory
+import org.gradle.api.tasks.Input
+import org.gradle.api.artifacts.Dependency
+
+import java.nio.file.Files
+
+class ListProjectDependencies extends DefaultTask {
+
+    @Input
+    String report = 'licenses.csv' //project.file("${project.buildDir}/reports/dependency-license/licenses.csv")
```
Comment (robbavey) [inline buildSrc/src/main/groovy/org/logstash/gradle/tooling/ListProjectDependencies.groovy:15]: Nit: leftover comment

**Review by robbavey — COMMENTED** [review]
Couple of questions, but in general, this looks good

File: buildSrc/src/main/groovy/org/logstash/gradle/tooling/ListProjectDependencies.groovy (hunk 1 above)
Comment (andsel) [reply buildSrc/src/main/groovy/org/logstash/gradle/tooling/ListProjectDependencies.groovy:25]: I've a doubt here if use a group name like `Logstash tooling` or use the punctuacted version.
It's just a string so don't use the package name is not a problem, my objective is to identify the Logstash's tooling tasks.
Which name to you think fit best?

Comment (robbavey) [reply buildSrc/src/main/groovy/org/logstash/gradle/tooling/ListProjectDependencies.groovy:25]: What you've got is fine, I think

**Review by robbavey — APPROVED** [review]
LGTM

**Pull request merged**